Cards/Effects/EffectHeal.gd

```diff
@@ -0,0 +1,15 @@
+class_name EffectHeal extends EffectBase
+## Heals an entity. [br]
+##
+## This effect will heal the target entity. The healing is calculated based on the value of the effect and the caster stats. [br]
+## Modifier key is HEAL. [br]
+
+## @Override [br]
+## Refer to [EffectBase]
+func apply_effect(caster: Entity, target: Entity, value: int) -> void:
+
+	# calculate modified heal given caster and target stats
+	var heal: float = EntityStats.get_value_modified_by_stats(GlobalEnums.PossibleModifierNames.HEAL, caster, target, value)
+	
+	# healing is essentially negative damage so we pass the var heal as negative value to make it "heal" the target
+	target.get_health_component().deal_damage(-heal, caster) 
```

Comment:
Kind of a nitpick here, but do we want to multiply the value calculated for the heal by -1 to make it a bit more readable? I feel adding `-` in that deal_damage may get missed.

Or better yet, we can make another function in the health component for heal_damage and do the negative then? 

```
var HEAL_MODIFIER = -1
var heal: float = EntityStats.get_value_modified_by_stats(GlobalEnums.PossibleModifierNames.HEAL, caster, target, value) * HEAL_MODIFIER
```

Reply:
My original plan was to have two different functions. Now that you mentioned it again, I wonder what should happen if due to a big number of negative modifiers, (For example, I want to heal 1 but I have like -10 in the healing stat) Should the player heal 0? Or should it cause damage?